src/modules/dashboard/components/RecoveryModeDialog/RecoveryModeDialogForm.tsx

```diff
@@ -0,0 +1,158 @@
+import Button from '~core/Button';
+import DialogSection from '~core/Dialog/DialogSection';
+import { Annotations } from '~core/Fields';
+import Heading from '~core/Heading';
+import Link from '~core/Link';
+import PermissionsLabel from '~core/PermissionsLabel';
+import PermissionRequiredInfo from '~core/PermissionRequiredInfo';
+
+import { useLoggedInUser, Colony } from '~data/index';
+import { useTransformer } from '~utils/hooks';
+
+import { getAllUserRoles } from '../../../transformers';
+import { canEnterRecoveryMode } from '../../../users/checks';
+
+import { FormValues } from './RecoveryModeDialog';
+import styles from './RecoveryModeDialogForm.css';
+
+const MSG = defineMessages({
+  title: {
+    id: 'admin.RecoveryModeDialog.RecoveryModeDialogForm.title',
+    defaultMessage: 'Enter Recovery mode',
+  },
+  recoveryModeDescription: {
+    id:
+      'admin.RecoveryModeDialog.RecoveryModeDialogForm.recoveryModeDescription',
+    defaultMessage: `If you believe that something dangerous is happening in
+    your colony (e.g. it is under attack), recovery mode will disable the colony
+    and prevent further activity until the issue has been overcome.`,
+  },
+  leavingRecoveryModeDescription: {
+    id:
+      // eslint-disable-next-line max-len
+      'admin.RecoveryModeDialog.RecoveryModeDialogForm.leavingRecoveryModeDescription',
+    defaultMessage: `
+    Leaving recovery requires the approval of a majority of members
+    holding the {roleRequired} permission. <a>Learn more.</a>`,
+  },
+  annotation: {
+    id: 'dashboard.RecoveryModeDialog.RecoveryModeDialogForm.annotation',
+    defaultMessage:
+      'Explain why you’re putting this colony into recovery mode (optional)',
+  },
+  noPermission: {
+    id: 'dashboard.RecoveryModeDialog.RecoveryModeDialogForm.noPermission',
+    defaultMessage: `You do not have the {roleRequired} permission required
+      to take this action.`,
+  },
+});
+
+interface Props {
+  back: () => void;
+  colony: Colony;
+}
+
+const RecoveryModeDialogForm = ({
+  back,
+  colony,
+  handleSubmit,
+  isSubmitting,
+}: Props & FormikProps<FormValues>) => {
+  const { walletAddress, username, ethereal } = useLoggedInUser();
+
+  const allUserRoles = useTransformer(getAllUserRoles, [colony, walletAddress]);
+
+  const hasRegisteredProfile = !!username && !ethereal;
+
+  const userHasPermission =
+    hasRegisteredProfile && canEnterRecoveryMode(allUserRoles);
+
+  return (
+    <>
+      <DialogSection appearance={{ theme: 'heading' }}>
+        <Heading
+          appearance={{ size: 'medium', margin: 'none' }}
+          text={MSG.title}
+          className={styles.title}
+        />
+      </DialogSection>
+      {!userHasPermission && (
+        <DialogSection>
+          <PermissionRequiredInfo requiredRoles={[ColonyRole.Recovery]} />
+        </DialogSection>
+      )}
+      <DialogSection>
+        <FormattedMessage {...MSG.recoveryModeDescription} />
+      </DialogSection>
+      <DialogSection>
```

Comment:
Please remove the padding from these two sections  _(you can use the `sidePadding` theme for that)_

And add manual padding to the paragraph elements. That way you can better control the spacing between the two text blocks, so it aligns more with the Figma designs

_(Basically, the spacing between the two is too large)_

Reply:
I left the padding for the first description section, and removed it from the second, then adding only the bottom one. Hope this approach is ok.